src/dataframe/dataframe.jl

```diff
@@ -239,7 +239,10 @@ ncol(df::DataFrame) = length(index(df))
 ##############################################################################
 
 # df[SingleColumnIndex] => AbstractVector, the same vector
-function Base.getindex(df::DataFrame, col_ind::ColumnIndex)
+Base.@propagate_inbounds Base.getindex(df::DataFrame, col_ind::Union{Signed, Unsigned}) =
```

Comment:
`Union{Signed, Unsigned}` is unfortunately needed instead of `Integer` because we want to keep compatibility with Julia 0.7 where handling of `Bool` indexing is different than on Julia 1.0.